### PR TITLE
Modified UV mappings in voxel_mesher to accomodate directional textures

### PR DESCRIPTION
**Pull request by Tintow**
Modified vertex and UV mappings to correctly orient textures that are sensitive to orientation.

Also added filter in raylib_premake5.lua to allow compilation on Linux

### CODE REVIEW

File: raylib_premake5.lua [hunk 1]
```diff
@@ -17,6 +17,10 @@ function platform_defines()
     filter {"system:macosx"}
         disablewarnings {"deprecated-declarations"}
 
+    filter {"system:linux"}
+        defines {"_GLFW_X11"}
+        defines {"_GNU_SOURCE"}
+
```
Comment (JeffM2501) [inline raylib_premake5.lua:23]: This change does not look required for the UVs.

File: voxel_mesher/main.cpp [hunk 2]
```diff
@@ -118,94 +118,94 @@ class CubeGeometryBuilder
...
 
-			SetSetUV(uvRect.x, uvRect.height);
-			PushVertex(position, 0, 1, 0);
+            SetSetUV(uvRect.width, uvRect.y);
+            PushVertex(position, 0, 1, 0);
 
-			SetSetUV(uvRect.width, uvRect.height);
-			PushVertex(position, 1, 1, 0);
+            SetSetUV(uvRect.x, uvRect.y);
+            PushVertex(position, 1, 1, 0);
```
Comment (JeffM2501) [inline voxel_mesher/main.cpp:137]: The whitespace here does not match the existing code, perhaps your tabs as spaces are not set correctly?

File: raylib_premake5.lua (hunk 1 above)
Comment (Tintow) [reply raylib_premake5.lua:23]: Okay that's my bad.  It doesn't compile on Linux in it's current form and I saw that this issue has been raised on a few other repos so I made the change but you're right, I should have made it a separate pull request.

File: voxel_mesher/main.cpp (hunk 2 above)
Comment (Tintow) [reply voxel_mesher/main.cpp:137]: That's strange.  I think all my editors are set to 4 space indentation which I think is correct looking at Raylib guidelines.  I'll review this and resubmit.  Thanks.

**Comment by Tintow** [comment]
Closing this request to make required changes before resubmitting.

**Comment by JeffM2501** [comment]
you can just make changes to your branch and it'll update this PR when you sync, no need for a new PR

**Comment by Tintow** [comment]
I can't re-open the pull request because I deleted the fork and started again.  My editor made so many changes to the original source because of things like spaces inside brackets etc. that it was a nightmare to try and extract just the required lines with the spacing as you need it.  It was just way easier to start again, branch it and just use a very basic text editor to make the changes and create a new PR.  Apologies if that is causing any issues your side - please let me know as I'm new to making PRs for other code bases so I'm learning from this.  Thanks.

**Pull request closed without merging**